### DISCUSSION TITLE
Search Results Custom Layout - Conditional elements

### DISCUSSION THREAD

**Discussion by llmedina** [microsoft-search/pnp-modern-search#1501] [Q&A] [answered]
I have been searching for a way to achieve something simple with Custom Layout templating: if a slot item value is zero-length, I don't wish to display anything

Example: 

Main Extension:  <--- if there is nothing in this field, I don't want to display the label.

I tried this: 
{{#if length (trim (slot item @root.slots.GreeterExtension)) 0}}
     <div></div>
{{else}}
     <div>Main Extension: {{slot item @root.slots.GreeterExtension}}</div>
{{/if}}

but the label still shows up with an empty GreeterExtension. What am I missing? Thanks in advance!

**Comment by FranckyC** [comment, marked as answer]
Hi @llmedina, this is likely an Handlebars issue. To compare a value you should use the `{{eq}}` operator instead of `if`. Something like this:

```
{{#eq (length(trim (slot item @root.slots.GreeterExtension))) 0}}
```

**Reply by llmedina** [reply]
Sweet! Thanks for the quick response @FranckyC ! That worked!
